src/Microsoft.AspNetCore.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -316,80 +318,90 @@ private void BuildFallbackBlock(TagHelperAttributeList attributes, TagHelperCont
                         var attribute = attributes[i];
                         if (!attribute.Name.Equals(SrcAttributeName, StringComparison.OrdinalIgnoreCase))
                         {
-                            var encodedKey = JavaScriptEncoder.Encode(attribute.Name);
-                            var attributeValue = GetAttributeValue(attribute.Value);
-                            var encodedValue = JavaScriptEncoder.Encode(attributeValue);
-
-                            AppendAttribute(builder, encodedKey, encodedValue, escapeQuotes: true);
+                            //var attributeValue = GetAttributeValueAsContent(attribute.Value);
+                            WriteAttribute(attribute.Name, attribute.Value, attribute.Structure, StringWriter);
                         }
                         else
                         {
                             addSrc = false;
-                            AppendEncodedVersionedSrc(attribute.Name, src, builder, generateForDocumentWrite: true);
+                            WriteEncodedVersionedSrc(attribute.Name, src, attribute.Structure, StringWriter);
                         }
                     }
 
                     if (addSrc)
                     {
-                        AppendEncodedVersionedSrc(SrcAttributeName, src, builder, generateForDocumentWrite: true);
+                        WriteEncodedVersionedSrc(SrcAttributeName, src, HtmlAttributeStructure.DoubleQuotedValue, StringWriter);
                     }
 
-                    builder.AppendHtml("><\\/script>");
+                    StringWriter.Write("></script>");
                 }
 
+                var stringBuilder = StringWriter.GetStringBuilder();
+                var scriptTags = stringBuilder.ToString();
+                stringBuilder.Clear();
+                var encodedScriptTags = JavaScriptEncoder.Encode(scriptTags);
+                builder.AppendHtml(encodedScriptTags);
+
                 builder.AppendHtml("\"));</script>");
             }
         }
 
-        private string GetAttributeValue(object value)
+        private IHtmlContent GetAttributeValueAsContent(object value)
```

Comment:
It's to make the comment on line 321 look more realistic :trollface:. Will remove.